Sources_v3/Database/DTOs/UserDTO_Tests.swift

```diff
@@ -174,4 +174,116 @@ class UserDTO_Tests: XCTestCase {
+            .map(\.createdAt)
+            .sorted(by: { $0 > $1 })
+        
+        let updatedDates = [payload1, payload2, payload3, payload4]
+            .map(\.updatedAt)
+            .sorted(by: { $0 > $1 })
+
+        // Save the users to DB. It doesn't matter which query we use because the filter for both of them is the same.
+        database.write { session in
+            try! session.saveUser(payload: payload1, query: queryWithCreatedAtSorting)
+            try! session.saveUser(payload: payload2, query: queryWithCreatedAtSorting)
+            try! session.saveUser(payload: payload3, query: queryWithCreatedAtSorting)
+            try! session.saveUser(payload: payload4, query: queryWithCreatedAtSorting)
+        }
+
+        // A fetch request with a createdAt sorting.
+        let fetchRequestWithCreatedAtSorting = UserDTO.userListFetchRequest(query: queryWithCreatedAtSorting)
+        // A fetch request with a updatedAt sorting.
+        let fetchRequestWithUpdatedAtSorting = UserDTO.userListFetchRequest(query: queryWithUpdatedAtSorting)
+
+        var usersWithCreatedAtSorting: [UserDTO] { try! database.viewContext.fetch(fetchRequestWithCreatedAtSorting) }
+        var usersWithUpdatedAtSorting: [UserDTO] { try! database.viewContext.fetch(fetchRequestWithUpdatedAtSorting) }
+
+        AssertAsync {
+            // Check the createdAt sorting.
+            Assert.willBeEqual(usersWithCreatedAtSorting.count, 4)
+            Assert.willBeEqual(usersWithCreatedAtSorting.map(\.userCreatedAt), createdDates)
+
+            // Check the updatedAt sorting.
+            Assert.willBeEqual(usersWithUpdatedAtSorting.count, 4)
+            Assert.willBeEqual(usersWithUpdatedAtSorting.map(\.userUpdatedAt), updatedDates)
+        }
+    }
+
+    /// `UserListSortingKey` test for sort descriptor and encoded value.
+    func test_channelListSortingKey() {
+        var channelListSortingKey = UserListSortingKey.createdAt
+        XCTAssertEqual(encodedUserListSortingKey(channelListSortingKey), "created_at")
+        XCTAssertEqual(
+            channelListSortingKey.sortDescriptor(isAscending: true),
+            NSSortDescriptor(key: "userCreatedAt", ascending: true)
+        )
+        
+        channelListSortingKey = .lastActiveAt
+        XCTAssertEqual(encodedUserListSortingKey(channelListSortingKey), "lastActiveAt")
+        XCTAssertEqual(
+            channelListSortingKey.sortDescriptor(isAscending: true),
+            NSSortDescriptor(key: "lastActivityAt", ascending: true)
+        )
+        
+        channelListSortingKey = .updatedAt
+        XCTAssertEqual(encodedUserListSortingKey(channelListSortingKey), "updated_at")
+        XCTAssertEqual(
+            channelListSortingKey.sortDescriptor(isAscending: true),
+            NSSortDescriptor(key: "userUpdatedAt", ascending: true)
+        )
+    }
+    
+    private func encodedUserListSortingKey(_ sortingKey: UserListSortingKey) -> String {
```

Comment:
The same function (although with `ChannelListSortingKey`) exists in `ChannelDTO_Tests` and `ChannelId_Tests` as `encode`, maybe we should provide this workaround via `JSONEncoder.stream` so when the times comes, we can remove it from a single place??

Reply:
Yep, thank you, makes sense. 
Fixed in 58110d6c.